public/app/features/alerting/unified/components/rules/RulesGroup.tsx

```diff
@@ -73,18 +81,32 @@ export const RulesGroup: FC<Props> = React.memo(({ group, namespace, expandAll }
     if (folderUID) {
       const baseUrl = `${config.appSubUrl}/dashboards/f/${folderUID}/${kbn.slugifyForUrl(namespace.name)}`;
       if (folder?.canSave) {
-        actionIcons.push(
-          <ActionIcon
-            aria-label="edit folder"
-            key="edit"
-            icon="pen"
-            tooltip="edit folder"
-            to={baseUrl + '/settings'}
-            target="__blank"
-          />
-        );
+        if (viewMode === 'grouped' && !isProvisioned) {
+          actionIcons.push(
+            <ActionIcon
+              aria-label="edit rule group"
+              data-testid="edit-group"
+              key="edit"
+              icon="pen"
+              tooltip="edit rule group"
+              onClick={() => setIsEditingGroup(true)}
+            />
+          );
+        }
+        if (viewMode === 'list') {
```

Comment:
I wonder if we could somehow distinguish these two actions. Maybe we could at least use a slightly different icon?

Reply:
I like the idea but unfortunately there's no viable alternative that I could find (ideally an "edit folder" icon)

Reply:
I've changed the icon and the action when you click it, it makes more sense this way 👍